runtime/src/lib.rs

```diff
@@ -44,3 +44,11 @@ pub unsafe extern "C" fn diplomat_alloc(size: usize, align: usize) -> *mut u8 {
 pub unsafe extern "C" fn diplomat_free(ptr: *mut u8, size: usize, align: usize) {
     alloc::alloc::dealloc(ptr, Layout::from_size_align(size, align).unwrap())
 }
+
+/// Whether a `&[u8]` is a `&str`
+/// # Safety
+/// - `ptr` and `size` must be a valid `&[u8]`
+#[no_mangle]
+pub unsafe extern "C" fn is_str(ptr: *const u8, size: usize) -> bool {
```

Comment:
question: is it possible to do this in pure C++ instead? would be nice to avoid the FFI call

Reply:
nit: the function should be prefixed with `diplomat_`

Reply:
I didn't find a C++ std function to validate UTF-8. Anyway, this feels safer, because passing `str::from_utf8` is the precondition for `str::from_utf8_unchecked`

Reply:
It seems useful to have this routine available to any backend that needs it. It shouldn't negatively impact ICU4X since we try hard to make most of our APIs work on potentially ill-formed UTF-8, which don't use this code path.